enip_backend/import_calls/run.py

```diff
@@ -0,0 +1,91 @@
+import json
+from datetime import datetime
+
+import boto3
+import psycopg2
+import pygsheets
+from pygsheets.authorization import service_account
+from pytz import timezone
+
+from ..enip_common.config import POSTGRES_URL, GSHEET_API_CREDENTIALS_SSM_PATH, CALLS_GSHEET_ID
+
+DATA_RANGE = ("A", "D")
+PUBLISH_DEFAULT_RANGE = ("F1", "G1")
+
+CALLED_AT_TZ = "US/Eastern"
+CALLED_AT_FMT = "%m/%d/%Y %H:%M:%S"
+
+
+def _get_gsheets_client():
+    ssm_param = boto3.client("ssm").get_parameter(
+        Name=GSHEET_API_CREDENTIALS_SSM_PATH, WithDecryption=True
+    )
+    param_value = ssm_param.get("Parameter", {}).get("Value")
+    credentials = service_account.Credentials.from_service_account_info(
+        json.loads(param_value),
+        scopes=("https://www.googleapis.com/auth/spreadsheets",)
+    )
+    return pygsheets.authorize(custom_credentials=credentials)
+
+
+def _convert_sheet_rows_to_dicts(vals):
+    header = vals[0]
+    return [{header[i]: row[i] for i in range(len(header))} for row in vals[1:]]
+
+
+def _handle_ap_called_at(value):
+    if not value:
+        return None
+    return datetime.strptime(value, CALLED_AT_FMT).replace(tzinfo=timezone(CALLED_AT_TZ))
+
+
+def _process_worksheet(sheet, worksheet_title):
+    wks = sheet.worksheet_by_title(worksheet_title)
+    publish_default_cells = wks.get_values(*PUBLISH_DEFAULT_RANGE)[0]
+    if publish_default_cells[0] != "Default":
+        raise Exception("Worksheet {} does not match the expected format".format(worksheet_title))
+    default_is_publish = publish_default_cells[1] == "Publish"
+
+    def _handle_published(value):
+        if value == "Default":
+            return default_is_publish
+        return value == "Yes"
+
+    rows = _convert_sheet_rows_to_dicts(wks.get_values(*DATA_RANGE))
+    return [
+        {
+            # Throw a key error if the sheet does not have the expected headers
+            "state": row["State"],
+            "ap_call": row["AP Call"] or None,
```

Comment:
I'm not sure that we want to import the calls from the spreadsheet -- the idea is that we populate the `AP Call` and `AP Called At` columns using the AP data, and then we read back just the `Published` column from the spreadsheet.

Reply:
oh i was thinking we'd do AP -> Google sheets -> db to keep the data flow simple,

Reply:
Oh, ok -- that works. So do you want to merge this in as-is and then add the AP -> google sheets bit later?

Reply:
yeah that's what i was thinking, but i think i can do ap to sheets tomorrow, so no need to merge it

Reply:
I guess one advantage of going AP -> DB, DB -> sheets (just list of calls + call time), sheets -> DB (just whether we're publishing) is that if sheets goes down, we can publish calls directly from the database, which feels like a useful backup plan?